core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -343,6 +343,13 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
         }
       }
     }
+
+    // Warn against the use of deprecated configs
+    deprecatedConfigs.values.foreach { dc =>
```

Comment:
Since you're doing this only once during the app's lifetime (this method is only called from SparkContext AFAICT), you could simplify the code that tracks whether warns have been printed in `DeprecatedConfig`. But ok to not do it if you want to limit the scope of the change.